### PR TITLE
Adding pyautogen as optional dependency in pyproject.toml

### PR DESCRIPTION
**Pull request by hemanthsavasere**
**Please describe the purpose of this pull request.**
Fixing the bug related while setup of the repository.

**How to test**
Setup a fresh new project in any IDE, autogen dependency was not met. After running "poetry add pyautogen --optional" command dependency was resolved.

**Have you tested this PR?**
Have you tested the latest commit on the PR? If so please provide outputs from your tests.
Yes. After running "poetry add pyautogen" dependency was resolved. Autogen dependency was not found in pyproject.toml file

**Related issues or PRs**
Please link any related GitHub [issues](https://github.com/cpacker/MemGPT/issues) or [PRs]
N.A

**Is your PR over 500 lines of code?**
If so, please break up your PR into multiple smaller PRs so that we can review them quickly, or provide justification for its length.
No

**Additional context**
Added

### CODE REVIEW

**Comment by hemanthsavasere** [comment]
@cpacker can you please review this ?

**Review by cpacker — CHANGES_REQUESTED** [review]
Can you fix the formatting issues? Check the contributor guide for running black

File: pyproject.toml [hunk 1]
```diff
@@ -54,12 +54,14 @@ uvicorn = {version = "^0.24.0.post1", optional = true}
 chromadb = "^0.4.18"
 pytest-asyncio = {version = "^0.23.2", optional = true}
 pydantic = "^2.5.2"
+pyautogen = {version = "0.2.0", optional = true}
```
Comment (cpacker) [inline pyproject.toml:57]: Just confirming - did you generate this + the lock file by doing:

```
poetry add pyautogen==0.2.0
```

Comment (hemanthsavasere) [reply pyproject.toml:57]: yes, this entry was addded by poetry

**Comment by hemanthsavasere** [comment]
> Can you fix the formatting issues? Check the contributor guide for running black
Yes, ran the command `poetry run black -l 140 .` but it has introduced many changes in other files that my current PR is not associated with.

**Comment by cpacker** [comment]
Seems like the code formatter messed up the code diff - I made a quick PR and merged it in here #https://github.com/cpacker/MemGPT/pull/616 and added you as a co-author to it.

**Pull request closed without merging**